File: src/leetcode/2026/01/Solution541.cpp
#include <string>
#include <iostream>
using namespace std;

class Solution541 {
public:
    string reverseStr(string s, int k) {
        for (int i = 0; i < s.size(); i += 2 * k) {
            int left = i;
            int right = min(i + k - 1, (int)s.size() - 1);
            while (left < right) {
                swap(s[left], s[right]);
                left++;
                right--;
            }
        }
        return s;
    }
};

int main() {
    Solution541 solution;

    string s1 = "abcdefg";
    int k1 = 2;
    cout << solution.reverseStr(s1, k1) << endl; // Expected output: "bacdfeg"

    string s2 = "abcd";
    int k2 = 2;
    cout << solution.reverseStr(s2, k2) << endl; // Expected output: "bacd"

    return 0;
}

File: src/leetcode/2026/01/Solution541.java
public class Solution541 {
    public String reverseStr(String s, int k) {
        char[] chars = s.toCharArray();
        for (int i = 0; i < chars.length; i += 2 * k) {
            int left = i;
            int right = Math.min(i + k - 1, chars.length - 1);
            while (left < right) {
                char temp = chars[left];
                chars[left] = chars[right];
                chars[right] = temp;
                left++;
                right--;
            }
        }
        return new String(chars);
    }

    public static void main(String[] args) {
        Solution541 solution = new Solution541();

        String s1 = "abcdefg";
        int k1 = 2;
        System.out.println("The reversed string of '" + s1 + "' with k = " + k1 + " is: " +
                solution.reverseStr(s1, k1));

        String s2 = "abcd";
        int k2 = 2;
        System.out.println("The reversed string of '" + s2 + "' with k = " + k2 + " is: " +
                solution.reverseStr(s2, k2));
    }
}
